### PR TITLE
docs: [E2E] Update readme for e2e tests

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

This PR updates the E2E testing instructions in our README to enhance clarity and usability. The updates focus on:

-  Simplified Test Build Preparation: Instructions have been added for easier and quicker preparation of test builds: use `yarn download-builds:test` to download pre-built test builds.
- Guidance on specific types of test builds: Clear instructions are now provided for running tests on different build types, ensuring contributors can accurately test against their desired build types.
- Add instructions to run e2e tests against specific feature flag
-  Remove the manual Chrome driver update process from readme, as our E2E tests now automatically download the latest Chrome driver.
- Updated the test path for an example.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2187

## **Manual testing steps**

Try to follow readme to run e2e tests to see if it's clear.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,7 +148,7 @@ Single e2e tests can be run with `yarn test:e2e:single test/e2e/tests/TEST_NAME.
 ```
 
 For example, to run the `account-details` tests using Chrome, with debug logging and with the browser set to remain open upon failure, you would use:
-`yarn test:e2e:single test/e2e/tests/account-details.spec.js --browser=chrome --debug --leave-running`
+`yarn test:e2e:single test/e2e/tests/account-menu/account-details.spec.js --browser=chrome --debug --leave-running`
```
Comment (DDDDDanica) [inline README.md:151]: Shall we add `multichain` one as well, as `MULTICHAIN=1 yarn test:e2e:chrome xxx`

Comment (chloeYue) [reply README.md:151]: Thanks Danica, that's a great suggestion. I think it's more accurate to enable feature flag when we create test build. I pushed a new section about how to run e2e tests against feature flag. Please check the new commit: 
https://github.com/MetaMask/metamask-extension/pull/23309/commits/6ec8bc8ea4f1bd22325696e8733360e6292bcc6c

File: README.md [hunk 2]
```diff
@@ -109,15 +109,25 @@ For Jest debugging guide using Node.js, see [docs/tests/jest.md](docs/tests/jest
...
+Before running e2e tests, you'll need a test build. You have two options:
 
-These test scripts all support additional options, which might be helpful for debugging. Run the script with the flag `--help` to see all options.
+1. Use `yarn download-builds:test` to quickly download and unzip test builds for Chrome and Firefox into the `./dist/` folder.
+2. Create a custom test build: for testing against different build types, use `yarn build:test`. This command allows you to generate test builds for various types, including:
+    - `yarn build:test` for main build
+    - `yarn build:test:flask` for flask build
+    - `yarn build:test:mmi` for mmi build
+    - `yarn build:test:mv3` for mv3 build
+
```
Comment (seaona) [inline README.md:124]: should we mentioned a 3rd option too: `yarn start:test` ?

**Comment by seaona** [comment]
Overall looks good :heart_eyes_cat:  Left a small suggestion

File: README.md (hunk 2 above)
Comment (chloeYue) [reply README.md:131]: Thanks Mariona, great catch, i have added it. Please check.

**Review by DDDDDanica — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
Looks good!  Thank you for improving documentation!

**Pull request merged**